modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItRunningQueriesSystemViewTest.java

```diff
@@ -0,0 +1,356 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine;
+
+import static org.apache.ignite.internal.TestWrappers.unwrapIgniteImpl;
+import static org.apache.ignite.internal.sql.engine.registry.RunningQueriesRegistryImpl.SCRIPT_QUERY_TYPE;
+import static org.apache.ignite.internal.sql.engine.util.SqlTestUtils.assertThrowsSqlException;
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.await;
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.waitForCondition;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasLength;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.internal.hlc.ClockService;
+import org.apache.ignite.internal.sql.SqlCommon;
+import org.apache.ignite.internal.sql.engine.registry.QueryExecutionPhase;
+import org.apache.ignite.internal.sql.engine.util.MetadataMatcher;
+import org.apache.ignite.internal.tx.InternalTransaction;
+import org.apache.ignite.lang.ErrorGroups.Sql;
+import org.apache.ignite.sql.ColumnType;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * End-to-end tests to verify {@code SQL_QUERIES} system view.
+ */
+public class ItRunningQueriesSystemViewTest extends BaseSqlMultiStatementTest {
+    @BeforeAll
+    void beforeAll() {
+        await(systemViewManager().completeRegistration());
+
+        sql("CREATE TABLE test(id INT PRIMARY KEY)");
+    }
+
+    @AfterEach
+    void cleanup() throws InterruptedException {
+        sql("DELETE FROM test");
+
+        checkNoPendingQueries();
+    }
+
+    @Override
+    protected int initialNodes() {
+        return 2;
+    }
+
+    @Test
+    public void singleStatement() {
+        String query = "SELECT * FROM SQL_QUERIES ORDER BY START_TIME";
+
+        // Verify metadata.
+        assertQuery(query)
+                .withDefaultSchema("SYSTEM")
+                .columnMetadata(
```

Comment:
will be good to add check nullability for all columns

Reply:
We can't change `nullability` for system view columns (they all `nullable` now, except "node name" column for node view).
I added these checks in test.